tests/unit/test_cluster.py

```diff
@@ -63,6 +67,18 @@ def test_bootstrap_cluster(
         _render_postgresql_conf_file.assert_called_once()
         _start_cluster.assert_called_once()
 
+    @patch("charm.PostgresqlCluster._change_owner")
+    @patch("os.chmod")
+    @patch("os.makedirs")
+    def test_create_directory(self, _makedirs, _chmod, _change_owner):
+        path = "/tmp/fakedir"
+        mode = 0o755
+
+        self.cluster._create_directory(path, mode)
+        _makedirs.assert_called_once_with(path, mode=mode, exist_ok=True)
+        _chmod.assert_called_once_with(path, mode)
+        _change_owner.assert_called_once_with(path)
```

Comment:
is it worth having this unittest, if all the lines of the function are mocked? i was faced with a similar situation in the mysql operator, and i realized that it will likely just be covered with functional tests

Reply:
You're Shayan. This test doesn't add much value. I removed it in favor of the functional tests.